cmd/snap/cmd_run.go

```diff
@@ -125,6 +145,18 @@ func maybeWaitForSecurityProfileRegeneration(cli *client.Client) error {
 		logger.Debugf("SystemKeyMismatch returned an error: %v", err)
 	}
 
+	// We have a mismatch but maybe it is only because systemd is shutting down
+	// and core or snapd were already unmounted and we failed to re-execute.
+	// For context see: https://bugs.launchpad.net/snapd/+bug/1871652
+	stopping, err := isStopping()
+	if err != nil {
+		logger.Debugf("cannot check if system is stopping: %s", err)
+	}
+	if stopping {
```

Comment:
Should this be:
```
if err != nil {
    logger.Debugf("cannot check if system is stopping: %s", err)
    stopping = true
}
```

Reply:
or rather than setting stopping, `return nil`

Reply:
I was thinking about this for a while but I have one observation that leans to "no".

14.04 doesn't have `systemctl is-system-running` - if we were always ignoring the system key we would have a regression of behavior there.